test/badge.js

```diff
@@ -674,3 +675,225 @@ export async function deleteBadgesOverApi(n) {
     for(let i = 0; i < n; i++)
         await deleteBadgeOverApi();
 }
+
+/**
+ * Downloads the JSON file of a badge from the awarded badge details page.
+ * @returns The content of the json file as string
+ * @param {import('selenium-webdriver').ThenableWebDriver} driver
+ */
+const downloadBadgeJson = async (driver, badgeName) => {
+  const overflowMenu = await driver.findElement(
+    By.css('button:has(svg[icon="icon_more"])')
+  );
+  await overflowMenu.click();
+
+  const downloadButton = await driver.findElement(
+    ExtendedBy.tagWithText('button', `Download JSON-Datei (${currentBadgeStandardVersion})`)
+  );
+  await downloadButton.click();
+
+  await waitForDownload(driver, new RegExp(/^\d{4}-\d{2}-\d{2}-[a-zA-Z0-9_ ]+\.json$/));
+
+  const now = new Date();
+  const badgeJsonName =
+    `${now.getFullYear()}-` +
+    `${('0' + (now.getMonth()+1)).slice(-2)}-` +
+    `${('0' + (now.getDate())).slice(-2)}-` +
+    `${badgeName.replace(' ', '_')}.json`;
+  const file = fs.readFileSync(`${downloadDirectory}/${badgeJsonName}`, { encoding: 'utf-8' });
+  return file;
+};
+
+/**
+ * Checks if the badge is conform with the currently implemented
+ * open badges version and downloads the badges JSON file
+ * to check its conformity.
+ * This assumes that the driver already navigated to the badge detail page
+ * of an awarded badge.
+ * @param {import('selenium-webdriver').ThenableWebDriver} driver
+ */
+export const validateBadgeVersion = async (driver) => {
+  const badgeStandardText = await driver.findElement(
+      ExtendedBy.tagWithText("dt", "Badge-Standard")
+    );
+  const badgeStandardVersion = await driver.findElement(
+    ExtendedBy.sibling(badgeStandardText, By.css("dd"))
+  );
+  const badgeStandardVersionText = await badgeStandardVersion.getText();
+  assert.equal(badgeStandardVersionText, currentBadgeStandardVersion);
+
+  const file = await downloadBadgeJson(driver, testBadgeTitle);
+  
+  const badgeAsJson = JSON.parse(file);
+  assert.equal(typeof badgeAsJson['@context'] === "string", false);
+  // The existence of the /credentials/ part in the context urls are
+  // unique for v3
+  assert.equal(badgeAsJson['@context'].reduce((prev, curr) => {
+    return prev || curr.indexOf('/credentials/') >= 0
```

Comment:
- You don't need the `{ return ...}` here; if I'm not mistaken you could also have `=> ...` directly
- Also I do not understand what's happening here. Maybe it's possible to write it in an easier way? *Maybe* you're trying to figure out if any instance contains `/credentials`? Then you could also use `.some(x => x.indexOf('/credentials') >= 0)`. If you wanted to make sure it's unique, I don't think you're current solution would work; for that you could for example use `.filter(x => x.indexOf('/credentials') >= 0).length == 1`
- Thirdly, you also don't need `assert.equal(..., true)`; you could just use `assert(...)`

Reply:
Don't know what I was thinking. The solution using `some` is so much simpler :facepalm: 
I added some context to the comment as well.